### PR TITLE
Added some diagrams to the development page

### PR DESCRIPTION
**Pull request by nhakmiller**
## Background

I created a couple of diagrams detailing different high level things about Panther. I couldn't figure out which ones were my favorite so I've just added them all here. Feel free to offer suggestions on cutting/re-organizing some as needed.

## Changes

- Added three new diagrams

## Testing

- Viewed in my IDE markdown renderer

### CODE REVIEW

File: docs/gitbook/development.md [hunk 1]
```diff
@@ -105,3 +113,23 @@ Since the majority of Panther is written in Go, the repo follows the standard [G
...
+
+Here is a slightly more detailed version of essentially the same architecture diagram as above.
+
+![Architecture diagram](.gitbook/assets/detailed-arch-diagram.png)
+
+While more detailed than the overview above, this diagram also simplifies some implementation details for clarity. For example, the majority of lambdas are not invoking each other directly but instead communicating via SQS Queues or DynamoDB streams. 
+
+### Data flow diagram
+
+This diagram focuses on where and how your data is being stored and processed, as opposed to which systems are interacting and why.
```
Comment (jacknagz) [inline docs/gitbook/development.md:131]:
```suggestion
This diagram shows where and how your data is stored and processed:
```

File: docs/gitbook/development.md [hunk 2]
```diff
@@ -105,3 +113,23 @@ Since the majority of Panther is written in Go, the repo follows the standard [G
...
+
+![Data flow diagram](.gitbook/assets/data-flow-diagram.png)
+
+It is important to note here (as the legend says) that the arrows are indicating the direction that data is being transferred, as opposed to the previous diagrams where arrows are indicating the direction that communication is being initiated from.
```
Comment (jacknagz) [inline docs/gitbook/development.md:135]:
```suggestion
The above arrows indicate the direction in which data is transferred, as opposed to the previous diagrams where arrows are indicating the direction that communication is being initiated.
```

File: docs/gitbook/development.md [hunk 3]
```diff
@@ -4,6 +4,14 @@ Panther is a collection of serverless applications deployed within your AWS acco
 
 The sections below provide guidance on how to extend Panther to meet your individual needs.
 
+## Architecture Diagram
+
+This diagram provides an overview of the core components of Panther, and how they are connected.
+
+![High level architecture diagram](.gitbook/assets/high-level-arch-diagram.png)
+
+This diagram has been simplified to just the core components of Panther. For a slightly more detailed architecture diagram, see the bottom of this page.
```
Comment (jacknagz) [inline docs/gitbook/development.md:13]:
```suggestion
For a more detailed architecture diagram, see the bottom of this page.
```

File: docs/gitbook/development.md [hunk 4]
```diff
@@ -105,3 +113,23 @@ Since the majority of Panther is written in Go, the repo follows the standard [G
 | [**pkg**](https://github.com/panther-labs/panther/tree/master/pkg)  | Standalone Go libraries that could be directly imported by other projects |
 | [**tools**](https://github.com/panther-labs/panther/tree/master/tools)  | Magefile source and other build infrastructure  |
 | [**web**](https://github.com/panther-labs/panther/tree/master/web)   | Source for the Panther web application  |
+
+## Additional Diagrams
+
+Here are some additional diagrams to supplement the high level architecture diagram above.
```
Comment (jacknagz) [inline docs/gitbook/development.md:119]:
```suggestion
The diagrams below can be used to understand Panther's architecture at a deeper level and provide insight into data flows.
```

File: docs/gitbook/development.md [hunk 5]
```diff
@@ -105,3 +113,23 @@ Since the majority of Panther is written in Go, the repo follows the standard [G
...
+
+### Detailed architecture diagram
```
Comment (jacknagz) [inline docs/gitbook/development.md:121]: Let's be consistent about title case here

File: docs/gitbook/development.md [hunk 6]
```diff
@@ -105,3 +113,23 @@ Since the majority of Panther is written in Go, the repo follows the standard [G
...
+
+Here is a slightly more detailed version of essentially the same architecture diagram as above.
```
Comment (jacknagz) [inline docs/gitbook/development.md:123]:
```suggestion
This diagram provides additional detail to the high-level diagram above:
```

**Comment by nhakmiller** [comment]
> Great work! The images seem small, but that might just be the way github is rendering them

Odd, for me they are rendering quite massive and I was thinking I may need to shrink them! I suspect GitHub is just doing some weirdness, if it looks bad after being published we can adjust.

**Comment by nhakmiller** [comment]
> These diagrams will be a big improvement.
> 
> Should we add an icon for Snowflake integration (optional)?

It's a good idea, but I think I will hold off for now until that relationship is more clearly defined (unless it is well defined and maybe I just don't understand it yet, feel free to ping me if you want to discuss)

**Pull request merged**